lib/image_edit_screen.dart

```diff
@@ -0,0 +1,219 @@
+      children: [
+        ElevatedButton(
+          child: const Text('Cancel Upload'),
+          onPressed: cancelImage,
+        ),
+        ElevatedButton(
+          child: const Text('Confirm Image'),
+          onPressed: cropImage,
+        )
+      ],
+    );
+  }
+
+  Widget getMenu() {
+    return Row(
+      children: <Widget>[
+        IconButton(
+          icon: const Icon(Icons.undo),
+          tooltip: 'Undo',
+          onPressed: () {
+            controller.rotation = 0;
+            controller.scale = 1;
+            controller.offset = Offset.zero;
+            setState(() {
+              rotation = 0;
+            });
+          },
+        ),
+        Expanded(
+          child: SliderTheme(
+            data: ThemeData(primarySwatch: Colors.blue).sliderTheme,
+            child: Slider(
+              divisions: 360,
+              value: rotation,
+              min: -180,
+              max: 180,
+              label: '$rotation°',
+              onChanged: (n) {
+                setState(() {
+                  rotation = n.roundToDouble();
+                  controller.rotation = rotation;
+                });
+              },
+            ),
+          ),
+        ),
+        PopupMenuButton<double>(
+          icon: const Icon(Icons.aspect_ratio),
+          itemBuilder: (context) => [
+            const PopupMenuItem(
+              child: Text("Original"),
+              value: 1000 / 667.0,
+            ),
+            const PopupMenuDivider(),
+            const PopupMenuItem(
+              child: Text("16:9"),
+              value: 16.0 / 9.0,
+            ),
+            const PopupMenuItem(
+              child: Text("4:3"),
+              value: 4.0 / 3.0,
+            ),
+            const PopupMenuItem(
+              child: Text("1:1"),
+              value: 1,
+            ),
+            const PopupMenuItem(
+              child: Text("3:4"),
+              value: 3.0 / 4.0,
+            ),
+            const PopupMenuItem(
+              child: Text("9:16"),
+              value: 9.0 / 16.0,
+            ),
+          ],
+          tooltip: 'Aspect Ratio',
+          onSelected: (x) {
+            controller.aspectRatio = x;
+            setState(() {});
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget getImage() {
+    final screenWidth = MediaQuery.of(context).size.width;
+    final screenHeight = MediaQuery.of(context).size.height;
+    final path = widget.imagePath;
+
+    return ConstrainedBox(
+      constraints: BoxConstraints(
+        maxWidth: 0.9 * screenWidth,
+        maxHeight: 0.7 * screenHeight,
+      ),
+      child: Crop(
+        onChanged: (decomposition) {
+          if (rotation != decomposition.rotation) {
+            setState(() {
+              rotation = ((decomposition.rotation + 180) % 360) - 180;
+            });
+          }
+        },
+        controller: controller,
+        shape: shape,
+        child: Image.asset(path, fit: BoxFit.cover),
```

Comment:
Since the image is a file from the camera and not an asset in the local folders, you need to use Image.file() not Image.asset(). Otherwise, there is an error in displaying the image.

Image.file(File(path), fit: BoxFit.cover)